sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/transformer/SparkFlattenDfTransformer.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2024 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.action.spark.transformer
+
+import com.typesafe.config.Config
+import io.smartdatalake.config.SdlConfigObject.{ActionId, DataObjectId}
+import io.smartdatalake.config.{FromConfigFactory, InstanceRegistry}
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.workflow.ActionPipelineContext
+import io.smartdatalake.workflow.action.generic.transformer.{GenericDfTransformer, SparkDfTransformer}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.{col, explode_outer, get}
+import org.apache.spark.sql.types.{ArrayType, DataType, StructType}
+
+import scala.annotation.tailrec
+
+/**
+ * Returns a flattened Dataframe from another DataFrame
+ * with a nested schema. Schemas in the form:
+ *
+ * |---parent
+ *
+ * |------|child1
+ *
+ * |------|child2
+ *
+ * are transformed to the following flat columns:
+ *
+ * |---parent_child1
+ *
+ * |---parent_child2
+ * @param name         name of the transformer
+ * @param description  Optional description of the transformer
+ * @param enableExplode Enables exploding Arrays in the Dataframe, therefore potentially increasing the number of rows. Default is set to "true"
+ */
+case class SparkFlattenDfTransformer(override val name: String = "sparkFlattenDataFrame",
+                                     override val description: Option[String] = None,
+                                     enableExplode: Boolean = true) extends SparkDfTransformer {
```

Comment:
Eventually disable explode by default. Because from my perspective when I would use the FlattenDfTransfomer as a user I would not expect from the name of the transfomer that he also explodes my data. Instead giving the possiblity to enable it, would need a conscious decision by the user.